connect/src/wormhole.ts

```diff
@@ -391,15 +392,28 @@ export class Wormhole {
     emitter: UniversalAddress | NativeAddress<PlatformName>,
     sequence: bigint,
     retries: number = 5,
+    opts?: {
+      retryDelay?: number;
+      requestTimeout?: number;
+    },
   ): Promise<Uint8Array | undefined> {
```

Comment:
Is there a reason not to throw an error containing the status or some error details instead of masking it and just returning undefined? 
In some situations it may be useful for an sdk client to decide depending on that.

Reply:
No _good_ reason, the retry/timeout stuff here is also begging to be written better, maybe also touching the retry-er thing here: https://github.com/wormhole-foundation/connect-sdk/blob/develop/connect/src/protocols/retry.ts